### ISSUE TITLE
Performance regression - possibly related to network requests in 3.5.0

### ISSUE THREAD

**Issue by lukeapage** [cypress-io/cypress#5467]
### Current behavior:

3.5.0: (all examples the tests pass)

spec x with 6 tests - 2:59
spec y with 1 test - 0:20
spec z with 3 tests - 1:57

3.4.1:

spec x with 6 tests - 1:20
spec y with 1 test - 0:09
spec z with 3 tests - 0:45

### Steps to reproduce: (app code and test code)

I need help with reproducing (I'm a paying customer).

I cannot reproduce this on my very fast dev machine, it only happens on our slow build servers.

My suspicion is around network - I've added logging for network requests

3.5.0:
```
  --> GET /static/desktop-black-ltr.css 200 167ms 765.58kb
  --> GET /static/integrationTests-desktop/entry.js 200 3,088ms 6.56mb
```

3.4.1:
```
  --> GET /static/desktop-black-ltr.css 200 22ms 765.58kb
  --> GET /static/integrationTests-desktop/entry.js 200 1,115ms 6.56mb
```

this is serving the file from the same server running the test, running through localhost.

Can you suggest anything I can do to debug?

### Versions

Cypress 3.5.0
Windows 10

**Comment by lukeapage** [comment]
Interestingly the times weren't that great with 3.4.1. I found that by serving the static files gzipped (fast compression), we saved 20% in run time.
I then gzipped with 3.5.0 and it makes the effect of upgrading from 100% slower to 10% slower.
It seems to happen on large resources requested on a low end machine

**Comment by jennifer-shehane** [comment]
We do have performance tests to try to ensure there are no performance regressions, so it's likely something specific to your project or to those requests th

**Comment by jennifer-shehane** [comment]
Sorry, didn't mean to close, stupid touchpad on Mac.

Likely it is something specific with your project or those requests. Feel free to email us any reproducible examples also at support@cypress.io if it is something that can be shared privately, but yes - we'll have to have a way to reproduce this.

**Comment by jennifer-shehane** [comment]
Hey @lukeapage any update on this?

**Comment by lukeapage** [comment]
No - I’m actually on paternity leave so now upgrading to use the latest cypress is on a backlog and this is the only blocking issue, everything else is either fixed or worked around.
I suspect it is not a cypress bug, but just a consequence of say electron being upgraded - but we have a plan to investigate, it’s just a couple of days work and not prioritized super high right now

**Comment by lukeapage** [comment]
(Because locally in multiple attempts to reproduce, cypress is faster in 3.7.0 - but build machines are a lot slower and something strange is going on..)

**Comment by jennifer-shehane** [comment]
Ok, we'll have to close this issue as there is not enough information to reproduce the problem.

**Comment by lukeapage** [comment]
This was still broken in 4.0.2
But we upgraded to 4.2.0 and cypress went from 50% slower than 3.5.0 to 20% faster.. excellent

**Comment by jennifer-shehane** [comment]
Yes, this was likely addressed by our 4.1.0 release where we fixed a regression introduced in 3.8.3 here: https://github.com/cypress-io/cypress/issues/6659
